pkg/storage/chunk/chunk.go

```diff
@@ -92,13 +93,16 @@ func NewChunk(userID string, fp model.Fingerprint, metric labels.Labels, c prom_
 func ParseExternalKey(userID, externalKey string) (Chunk, error) {
 	if !strings.Contains(externalKey, "/") {
 		return parseLegacyChunkID(userID, externalKey)
-	}
-	chunk, err := parseNewExternalKey(userID, externalKey)
-	if err != nil {
-		return Chunk{}, err
-	}
+	} else if strings.Count(externalKey, "/") > 1 {
+		return parseNewerExternalKey(userID, externalKey)
+	} else {
```

Comment:
I've been there. May I suggest to add a key version prefix or suffix? E.g.
```
keyv1:<user id>/<fingerprint>:<start time>:<end time>:<checksum>
```
Unless you are really sure that the key schema will never change again.

Reply:
Responding to this great comment here: https://github.com/grafana/loki/pull/4857

Thanks!